src/main/java/com/google/devtools/build/lib/runtime/QueryRuntimeHelper.java

```diff
@@ -91,6 +108,34 @@ public void afterQueryOutputIsWritten() {}
       @Override
       public void close() {}
     }
+
+    private static class FileQueryRuntimeHelper implements QueryRuntimeHelper {
+      private final Path path;
+      private final OutputStream out;
+
+      public FileQueryRuntimeHelper(Path path) throws IOException {
+        out = path.getOutputStream();
+        this.path = path;
+      }
+
+      @Override
+      public OutputStream getOutputStreamForQueryOutput() {
+        return out;
```

Comment:
Not sure whether it would be preferable to open lazily here. I opened in the constructor to keep the code simple but am open to deferring it to speed up Bazel startup time or something.

Reply:
Can you use a supplier?

Reply:
How about this? LazyOutputStream defers opening until the first use, where all callers are listening for `IOException` anyway.

Reply:
TBH I'm not sure if you really need the LazyFileOutputStream here, the former might work just fine and we can optimize it later if necessary - unless you see evidence of slowdown otherwise?